dotnet/src/VectorDataIntegrationTests/PgVectorIntegrationTests/PostgresEmbeddingGenerationTests.cs

```diff
@@ -25,14 +25,36 @@ public override VectorStore CreateVectorStore(IEmbeddingGenerator? embeddingGene
         [
             services => services
                 .AddSingleton(PostgresTestStore.Instance.DataSource)
-                .AddPostgresVectorStore()
+                .AddPostgresVectorStore(),
+            services => services
+                .AddTransient<NpgsqlDataSource>(_ =>
+                {
+                    NpgsqlDataSourceBuilder builder = new (PostgresTestStore.Instance.ConnectionString);
+                    builder.UseVector();
+                    return builder.Build();
+                })
+                .AddPostgresVectorStore(lifetime: ServiceLifetime.Transient),
```

Comment:
this is an example of how our users can use `ServiceLifetime.Transient` with the existing overload